### PR TITLE
Set a flag for dev builds to fix manifest issue

### PR DESCRIPTION
**Pull request by wojtekn**
Related to https://github.com/Automattic/dotcom-forge/issues/7339

## Proposed Changes

I propose to set an environment variable to state that the given build is a dev build. That way, the manifest can be generated correctly for dev builds after recent pipeline changes.

## Testing Instructions

See if the manifest is generated correctly and if dev builds are updated.

Currently, 'Distribute Dev Builds' step runs only for trunk, so it can't be tested before the merge.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -71,6 +71,7 @@ steps:
       buildkite-agent artifact download "*.exe" .
 
       echo "--- :node: Generating Release Manifest"
+      export IS_DEV_BUILD=true
```
Comment (p-jackson) [inline .buildkite/pipeline.yml]: Or `IS_DEV_BUILD=true node ./scripts/generate-releases-manifest.mjs` to make it explicit the flag is specifically for the `generate-releases-manifest` script.

**Review by p-jackson — APPROVED** [review]
LGTM.

The step doesn't run on PR builds, so there's no better way to test other than to merge. YOLO.

**Comment by wojtekn** [comment]
It worked fine, as the 'dev' builds section in manifest was regenerated:

```
Version: 1.0.2
--
  | Is dev build: true
  | Current commit: 1fe21c8
  | Downloading current manifest ...
  | (node:1670) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
  |  
  | Download complete
  | Parsing current release info ...
  | Overriding latest dev release ...
  | Overwrote latest dev release
  | Done generating releases manifest.
```

And here is the section:

```
% curl https://cdn.a8c-ci.services/studio/releases.json
{
  "dev": {
    "darwin": {
      "universal": {
        "sha": "1fe21c8",
        "url": "https://cdn.a8c-ci.services/studio/studio-darwin-1fe21c8.app.zip"
      },
      "x64": {
        "sha": "1fe21c8",
        "url": "https://cdn.a8c-ci.services/studio/studio-darwin-x64-1fe21c8.app.zip"
      },
      "arm64": {
        "sha": "1fe21c8",
        "url": "https://cdn.a8c-ci.services/studio/studio-darwin-arm64-1fe21c8.app.zip"
      }
    },
    "win32": {
      "sha": "1fe21c8",
      "url": "https://cdn.a8c-ci.services/studio/studio-win32-1fe21c8.exe"
    }
  },
[...]
```

However, links don't work so we need to check it more.

**Comment by p-jackson** [comment]
I notice the `-universal-` bit is missing from the universal build's filename. But I don't know what's wrong with the other ones. Should the filename include `-v1.0.2-dev.1fe21c8`?

**Comment by wojtekn** [comment]
@p-jackson I fixed those in https://github.com/Automattic/studio/pull/153

**Pull request merged**